Add support for IW610

gangli02 opened the pull request:
Add Kconfig NXP_IW610, NXP_IW610_MURATA_2LL_M2, NXP_IW610_RD_USD.
Added seperate shield overlay for supported M.2 2LL module to enable BT or WIFI or Both.

gangli02 committed:
drivers: wifi: nxp: add Kconfig for IW610

Add Kconfig NXP_IW610, NXP_IW610_MURATA_2LL_M2,
NXP_IW610_RD_USD

Signed-off-by: Gang Li <gang.li_1@nxp.com>

nngt88 committed:
drivers: bluetooth: hci: nxp: add Kconfig for IW610

Add Kconfig BT_NXP_IW610.

Signed-off-by: Guotao Zhang <guotao.zhang@nxp.com>

kartben commented on boards/shields/nxp_m2_wifi_bt/Kconfig.shield:
add empty line at the end of the file

gangli02 replied:
Okay

kartben commented on boards/shields/nxp_m2_wifi_bt/nxp_m2_2ll_wifi_bt.overlay:
Nit: it would probably be better to use the following naming scheme for the shields nxp_m2_wifi_bt_XXX

gangli02 replied:
The name is consistent with other m2 shields. (nxp_m2_1xk_wifi_bt, nxp_m2_2el_wifi_bt)
m2_1xk, m2_2el, m2_2ll are types of m2 shield, maybe nxp_m2_XXX_wifi_bt or nxp_wifi_bt_m2_XXX are better, right?
@abhinavnxp @nirav-agrawal what do you think?

gangli02 committed:
manifest: update hal_nxp to add support for IW610

Add support for IW610

Signed-off-by: Gang Li <gang.li_1@nxp.com>

nirav-agrawal reviewed (commented):
Hi @gangli02, changes are ok for the Shield addition and for BT part, let other folks to review and approve.

dleach02 commented:
The open/close is to kick off the new action to get the new blobs label to apply

gangli02 committed:
boards: shields: nxp_m2_wifi_bt: add nxp_m2_2ll_wifi_bt module

Added seperate shield overlay for supported M.2 2LL module to enable
BT or WIFI or Both.

Signed-off-by: Gang Li <gang.li_1@nxp.com>

gangli02 commented:
> Hi @gangli02 , These Twister build errors should be resolved now with #91056 . Please rebase

OK, let me update

DerekSnell reviewed (approved):
Hi @gangli02 ,

The commit order is incorrect in this branch.  Commits that are dependencies for other commits should go first.  I would make the commits adding the shied support last, since they depend on the Kconfigs and HAL update in the other commits.  But I approve.

mmahadevan108 commented:
> Hi @gangli02 ,
> 
> The commit order is incorrect in this branch. Commits that are dependencies for other commits should go first. I would make the commits adding the shied support last, since they depend on the Kconfigs and HAL update in the other commits. But I approve.

@DerekSnell , I have updated the commit order.

mmahadevan108 commented:
@kartben, can you help take a look at this PR.
